src/main/java/org/terasology/metalrenegades/ai/actions/SimpleComponentTargetAction.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.metalrenegades.ai.actions;
+
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.behavior.BehaviorAction;
+import org.terasology.logic.behavior.core.Actor;
+import org.terasology.logic.behavior.core.BaseAction;
+import org.terasology.logic.behavior.core.BehaviorState;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.metalrenegades.ai.CitizenNeed;
+import org.terasology.metalrenegades.ai.component.SimpleSourceComponent;
+import org.terasology.minion.move.MinionMoveComponent;
+import org.terasology.registry.In;
+
+/**
+ * Sets the character's target to a source which fulfills the specified need type.
+ */
+@BehaviorAction(name = "set_target_component")
+public class SimpleComponentTargetAction extends BaseAction {
+
+    @In
+    private EntityManager entityManager;
+
+    private String needType;
+
+    @Override
+    public BehaviorState modify(Actor actor, BehaviorState result) {
+        CitizenNeed.Type needTypeValue = CitizenNeed.Type.valueOf(needType);
+
+        Vector3f closestLocation = new Vector3f(1000, 1000, 1000);
```

Comment:
While this is correct, it looks a bit odd to me. I would go with a float `maxDistSquared` instead, use it for comparison instead of computing the squared distance of this vector over and over again. You can then set this to `null` and later check for `null` instead of this float comparison. This would be cleaner, imho.

Reply:
Agreed, changed it and it looks much cleaner now 👍